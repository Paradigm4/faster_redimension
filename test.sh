#!/bin/bash

MYDIR=`dirname $0`
pushd $MYDIR > /dev/null
MYDIR=`pwd`
OUTFILE=$MYDIR/test.out
EXPFILE=$MYDIR/test.expected

rm -rf $OUTFILE > /dev/null 2>&1

iquery -anq "remove(foo)" > /dev/null 2>&1
iquery -anq "store(build(<a:double,b:string,c:int64,x:int64>[i=1:10,3,0], '[(1.1,a,0,0),(2.2,b,1,null),(3.3,c,null,2),(4.4,d,null,null),(5.5,f,4,3),(6.6,g,4,4),(7.7,h,4,5),(8.8,null,4,6),(9.9,i,0,7),(10.1,k,0,8)]', true), foo)" > /dev/null 2>&1

iquery -aq "faster_redimension(foo, <a:double>[c=0:*,1,0,x=0:*,1,0])" >> $OUTFILE 2>&1
iquery -aq "faster_redimension(foo, <a:double>[c=0:*,1,0,x=0:*,2,0])" >> $OUTFILE 2>&1
iquery -aq "faster_redimension(foo, <a:double, b:string>[c=0:*,3,0,x=0:*,2,0])" >> $OUTFILE 2>&1
iquery -aq "faster_redimension(foo, <a:double, b:string>[c=0:*,10,0,x=0:*,10,0])" >> $OUTFILE 2>&1
iquery -aq "faster_redimension(foo, <b:string>[x=0:*,10,0])" >> $OUTFILE 2>&1
iquery -aq "faster_redimension(foo, <b:string>[x=0:*,4,0])" >> $OUTFILE 2>&1

iquery -aq "sort(unpack(faster_redimension(foo, <b:string>[c=0:*,4,0, synthetic=5:*,10,0]), i), c,synthetic)" >> $OUTFILE 2>&1
iquery -aq "sort(unpack(faster_redimension(foo, <a:double, b:string>[synthetic=3:*,10,0, c=0:*,4,0]), i), c, synthetic)" >> $OUTFILE 2>&1

diff $OUTFILE $EXPFILE

