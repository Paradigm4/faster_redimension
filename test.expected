{c,x} a
{0,0} 1.1
{0,7} 9.9
{0,8} 10.1
{4,3} 5.5
{4,4} 6.6
{4,5} 7.7
{4,6} 8.8
{c,x} a
{0,0} 1.1
{0,7} 9.9
{0,8} 10.1
{4,3} 5.5
{4,4} 6.6
{4,5} 7.7
{4,6} 8.8
{c,x} a,b
{0,0} 1.1,'a'
{0,7} 9.9,'i'
{0,8} 10.1,'k'
{4,3} 5.5,'f'
{4,4} 6.6,'g'
{4,5} 7.7,'h'
{4,6} 8.8,null
{c,x} a,b
{0,0} 1.1,'a'
{0,7} 9.9,'i'
{0,8} 10.1,'k'
{4,3} 5.5,'f'
{4,4} 6.6,'g'
{4,5} 7.7,'h'
{4,6} 8.8,null
{x} b
{0} 'a'
{2} 'c'
{3} 'f'
{4} 'g'
{5} 'h'
{6} null
{7} 'i'
{8} 'k'
{x} b
{0} 'a'
{2} 'c'
{3} 'f'
{4} 'g'
{5} 'h'
{6} null
{7} 'i'
{8} 'k'
{n} c,synthetic,b
{0} 0,5,'a'
{1} 0,6,'i'
{2} 0,7,'k'
{3} 1,5,'b'
{4} 4,5,'f'
{5} 4,6,'g'
{6} 4,7,null
{7} 4,8,'h'
{n} synthetic,c,a,b
{0} 3,0,1.1,'a'
{1} 4,0,9.9,'i'
{2} 5,0,10.1,'k'
{3} 3,1,2.2,'b'
{4} 3,4,5.5,'f'
{5} 4,4,6.6,'g'
{6} 5,4,8.8,null
{7} 6,4,7.7,'h'
